spec/javascripts/components/story/expanded_story/expanded_story_task_spec.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import ExpandedStoryTask from 'components/story/ExpandedStory/ExpandedStoryTask';
+
+describe('<ExpandedStoryTask />', () => {
+
+  const setup = propOverrides => {
+    const defaultProps = {
+      story: { tasks: [] },
+      onDelete: sinon.spy(),
+      onUpdate: sinon.spy(),
+      onSave: sinon.spy(),
+      ...propOverrides
+    };
+
+    const wrapper = shallow(<ExpandedStoryTask {...defaultProps} />);
+    const wrapperInstance = wrapper.instance();
+    const button = wrapper.find('button');
+    const input = wrapper.find('input');
+
+    return { wrapper, wrapperInstance, button, input };
+  };
+
+  it('renders component title', () => {
+    const { wrapper } = setup();
+
+    expect(wrapper.text()).toContain(I18n.t('story.tasks'));
+  });
+
+  it('renders component content', () => {
+    const { wrapper } = setup();
+
+    expect(wrapper.exists('.Story__list-task checkbox')).toBe(true);
+    expect(wrapper.exists('.task_form clearfix')).toBe(true);
+    expect(wrapper.exists('.add-task btn btn-default btn-xs')).toBe(true);
+    expect(wrapper.exists('.form-control input-sm')).toBe(true);
+    expect(wrapper.exists('TasksList')).toBe(true);
+  });
+
+  describe('onHandleSubmit', () => {
+    it('calls onSave with a task', () => {
+      const task = 'New Task';
+      const onSaveSpy = sinon.spy();
+      const { wrapperInstance } = setup({ onSave: onSaveSpy });
+      wrapperInstance.setState({ task })
+
+      wrapperInstance.onHandleSubmit();
+
+      expect(onSaveSpy).toHaveBeenCalledWith(task);
+    });
+
+    it('calls setState with a empty task', () => {
+      const { wrapperInstance } = setup();
+      sinon.spy(wrapperInstance, 'setState');
```

Comment:
Avoid spying components methods like this, prefer to use this instead https://airbnb.io/enzyme/docs/api/ShallowWrapper/state.html